### PR TITLE
Modernise - now uses and works with latest formats and versions of everything

### PR DESCRIPTION
**Pull request by aholland**
Moved build from Build.scala in project subdir to build.sbt in root, getting rid of deprecated Project.defaultSettings etc. imports, etc.
***and***
updated library version for sbt, scala, slick, sl4j, h2 and the Eclipse plugin to latest stable versions as of the time of this commit. Also had to remove prefix "scala." from val profile in Example.scala.

### CODE REVIEW

**Comment by cvogt** [comment]
@aholland What do you like about build.sbt better than Build.scala? I always found it more confusing.

**Comment by aholland** [comment]
Quick response, thanks! It's simply a matter of using what I understand to be the current most orthodox  style of build. As described here: http://www.scala-sbt.org/0.13/docs/Basic-Def.html

Having read that page, I can understand that to someone who has not read that page but is familiar with Scala generally, a scala file may be less confusing. 

In part I'd like dive into the discussions and say that I don't think the three implied (i.e. automatic) imports and couple of other minor differences are really *that* confusing if you know about them, and I have become used to the layout of a normal build.sbt file which is less cluttered.  

But more importantly, I don't think we should decide on a project-by-project basis whether to do things the standard way. It simply is the new standard. You may feel more sympathetic if I tell you that in making a recent contribution in bringing the Play AutoRefresh plugin up to date, James Ward got me to change the style of the sbt back from the style I took from and old build example I was following, because I had unwittingly moved it to a *less* modern (and therefore in his view less correct) layout.

**Comment by aholland** [comment]
Note that circleci check failed due to circleci not having the latest version of sbt available to it.

**Comment by aholland** [comment]
It's late here in Africa - I look forward to your response in the morning.

File: project/build.properties [hunk 1]
```diff
@@ -1 +1 @@
-sbt.version=0.13.0
+sbt.version=0.13.11
```
Comment (cvogt) [inline project/build.properties]: 0.13.9 should work. How about we go with that one for now until circle updates?

File: build.sbt [hunk 2]
```diff
@@ -0,0 +1,31 @@
+val slickVersion = "3.1.1"
+
+lazy val mainProject = (project in file(".")).settings(
```
Comment (cvogt) [inline build.sbt]: does `build.sbt` still support the constructor syntax `Project(...` ? Let's stick with that as a compromise?

**Comment by cvogt** [comment]
Africa nice :) Where?

**Comment by aholland** [comment]
Changed to sbt 0.13.9 and Project() as requested. Have squashed the 3 commits into one that you can proceed if you're happy.  I'm here: https://goo.gl/maps/ekcAXGoqeF52 :)

**Comment by aholland** [comment]
Hi Jan. Was just reminded about this PR when I read in Heroku's docs that "Heroku recognizes an app as Scala by the existence of a build.sbt file in the root directory."!  This is a just a nudge to act on this PR. - Anthony

**Comment by cvogt** [comment]
also Build.scala is apparently being deprecated in sbt 1.0

**Comment by cvogt** [comment]
thx :)

**Pull request merged**